system-contracts/test/NonceHolder.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { expect } from "chai";
+import type { NonceHolder } from "../typechain";
+import { NonceHolderFactory } from "../typechain";
+import {
+  TEST_BOOTLOADER_FORMAL_ADDRESS,
+  TEST_DEPLOYER_SYSTEM_CONTRACT_ADDRESS,
+  TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS,
+} from "./shared/constants";
+import { prepareEnvironment, setResult } from "./shared/mocks";
+import { deployContractOnAddress, getWallets } from "./shared/utils";
+import { ethers, network } from "hardhat";
+
+describe("NonceHolder tests", () => {
+  const wallet = getWallets()[0];
+  let nonceHolder: NonceHolder;
+  let nonceHolderAccount: ethers.Signer;
+  let deployerAccount: ethers.Signer;
+
+  before(async () => {
+    await prepareEnvironment();
+    await deployContractOnAddress(TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS, "NonceHolder");
+    nonceHolder = NonceHolderFactory.connect(TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS, wallet);
+    nonceHolderAccount = await ethers.getImpersonatedSigner(TEST_BOOTLOADER_FORMAL_ADDRESS);
```

Comment:
Can you use another address in the test cases when it doesn't matter, it may be a bit confusing when you are using these system contract accounts

Reply:
done